lib/auth/clt.go

```diff
@@ -238,13 +238,13 @@ func (c *Client) GetTransport() *http.Transport {
 // PostJSON is a generic method that issues http POST request to the server
 func (c *Client) PostJSON(
 	endpoint string, val interface{}) (*roundtrip.Response, error) {
-	return httplib.ConvertResponse(c.Client.PostJSON(endpoint, val))
+	return httplib.ConvertResponse(c.Client.PostJSON(context.TODO(), endpoint, val))
```

Comment:
this is going to slow down every single request as it now has to allocate context on every operation. Create a context per client once.

Reply:
Please also make sure that this is not the case for all the other code here.